src/main/scala/io/spicelabs/goatrodeo/util/DotnetDetector.scala

```diff
@@ -66,10 +67,18 @@ object FileInputStreamEx {
         "metadata data needs the RESOURCE_NAME_KEY set to the path name"
       )
     val f = File(path)
-    if (!f.exists())
-      throw new IllegalArgumentException(
-        s"RESOURCE_NAME_KEY is set to ${path} but that path doesn't exist"
-      )
-    FileInputStreamEx(metadata.get(TikaCoreProperties.RESOURCE_NAME_KEY))
+    if (!f.exists()) {
+      if (is.isInstanceOf[TikaInputStream]) {
+        val tika = is.asInstanceOf[TikaInputStream]
+        FileInputStreamEx(tika.getPath().toString())
+      } else {
+        throw IllegalArgumentException(
```

Comment:
Please don't throw exceptions. Treating failures as either Option or Try is better Scala... plus most of the code doesn't do well with exceptions.